src/core/frames/frame_controller.ts

```diff
@@ -358,6 +358,7 @@ export class FrameController
 
   private navigateFrame(element: Element, url: string, submitter?: HTMLElement) {
     const frame = this.findFrameElement(element, submitter)
+    this.pageSnapshot = PageSnapshot.fromElement(element).clone()
```

Comment:
It'll end up assigning the same `Document` instance, but how do you feel about passing the `frame` instead of the `element` here?